### ISSUE TITLE
How to get the class name from PyToch Hub

### ISSUE THREAD

**Issue by shershunov** [ultralytics/yolov5#3473]
## ❔Question

How can I get only the class name of the detected object?

**Comment by glenn-jocher** [comment]
@EkelviNistars see YOLOv5 **PyTorch Hub** tutorial for getting class names of detections:

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://docs.ultralytics.com/yolov5/tutorials/model_export8)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [TorchScript, ONNX, CoreML Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

**Comment by shershunov** [comment]
The tutorial says how to print the class name, and I need to assign a name to a variable.

**Comment by glenn-jocher** [comment]
@EkelviNistars in python you can assign a value to a variable by using the equals sign:
```
x = 1
```

**Comment by shershunov** [comment]
Thank you so much, I really didn't know how to assign a value to a variable. (sarcasm)

**Comment by yakhyo** [comment]
I have read the PyTorch Hub tutorial but I could not find any clue how to get the class name from the predictions. Could you guide me how to get the class name (predicted class name) from the predictions. The output looks like this in my case: 
![image](https://user-images.githubusercontent.com/28424328/187142040-8d517f8d-c9ec-4540-93bf-1514a9594b3d.png)

**Comment by glenn-jocher** [comment]
@yakhyo 👋 Hello! Thanks for asking about **handling inference results**. YOLOv5 🚀 [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) models allow for simple model loading and inference in a pure python environment without using `detect.py`. 

### Simple Inference Example

This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://docs.ultralytics.com/yolov5/tutorials/model_export) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # yolov5n - yolov5x6 official model
#                                            'custom', 'path/to/best.pt')  # custom model

# Images
im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list

# Inference
results = model(im)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.
results.xyxy[0]  # im predictions (tensor)

results.pandas().xyxy[0]  # im predictions (pandas)
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie

results.pandas().xyxy[0].value_counts('name')  # class counts (pandas)
# person    2
# tie       1
```
<img width="550" src="https://user-images.githubusercontent.com/26833433/149679662-b2021c78-59b2-4f52-b768-1e5161537c85.png">

See [YOLOv5 PyTorch Hub Tutorial](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) for details.

Good luck 🍀 and let us know if you have any other questions!

**Comment by tannaom29** [comment]
![image](https://user-images.githubusercontent.com/86075925/211819307-aaf9ad7f-0981-492b-95e1-f71a0c1115d7.png)

**Comment by glenn-jocher** [comment]
@tannaom29 it looks like you are using the YOLOv5 `predict` method, which returns detection results with class labels, confidence scores, and bounding box coordinates. To access the predicted class name from the results, you can use the following code snippet:

```python
results = model(im)
class_name = results.names[int(results.xyxy[0][0][-1])]
print(class_name)
```

This code retrieves the class name associated with the first detected object in the `results`. The `int(results.xyxy[0][0][-1])` retrieves the class index, and `results.names` provides the class names.

For more details, you can refer to the YOLOv5 [PyTorch Hub Tutorial](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading).

Feel free to reach out if you have any further questions or need additional assistance!
